ludwig/schema/llms/model_parameters.py

```diff
@@ -0,0 +1,73 @@
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.error import ConfigValidationError
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.utils import ludwig_dataclass
+
+
+@DeveloperAPI
+@ludwig_dataclass
+class RoPEScalingConfig(schema_utils.BaseMarshmallowConfig):
+    """Dynamic RoPE-scaling (rotary position embeddings) to extend the context length of LLM like LLaMA, GPT-NeoX,
+    or Falcon.
+
+    This parameter is a dictionary containing the scaling configuration
+    for the RoPE embeddings. Currently supports three scaling strategies: linear and dynamic. Their
+    scaling factor must be an float greater than 1. The expected format is {'type': strategy name,
+    'factor': scaling factor}
+    """
+
+    def __post_init__(self):
+        if self.type and not self.factor:
+            raise ConfigValidationError(f"`rope_scaling`'s `factor` field must be an float > 1, got {self.factor}")
```

Comment:
The error message seems incongruous with the condition. Shouldn't this be something like "if a type is provided, you must specify a factor" or something?